x/staging/posts/keeper/grpc_query.go

```diff
@@ -16,98 +15,38 @@ import (
 
 var _ types.QueryServer = Keeper{}
 
-func (k Keeper) getPostResponse(ctx sdk.Context, post types.Post) types.QueryPostResponse {
-	// Get the reactions
-	postReactions := k.GetPostReactions(ctx, post.PostID)
-	if postReactions == nil {
-		postReactions = []types.PostReaction{}
-	}
-
-	// Get the children
-	childrenIDs := k.GetPostChildrenIDs(ctx, post.PostID)
-	if childrenIDs == nil {
-		childrenIDs = []string{}
-	}
-
-	//Get the user answers if poll exist
-	var answers []types.UserAnswer
-	if post.PollData != nil {
-		answers = k.GetUserAnswersByPost(ctx, post.PostID)
-	}
-
-	// Crete the response object
-	return types.QueryPostResponse{
-		Post:        post,
-		UserAnswers: answers,
-		Reactions:   postReactions,
-		Children:    childrenIDs,
-	}
-}
-
+// Posts implements the Query/Posts gRPC method
 func (k Keeper) Posts(goCtx context.Context, req *types.QueryPostsRequest) (*types.QueryPostsResponse, error) {
```

Comment:
I think that here we should require the user to give us a valid subspace identifier, blocking them from querying the whole set of posts. There should not be any reason to do that anyway.

Reply:
@RiccardoM Then, I think this query function should be renaming into `SubspacePosts`, instead.

Reply:
@dadamu I prefer we stick with `Posts` instead since there's no need to prepend the `Subspace` prefix as it doesn't add anything to the query. Also, it might confuse developers thinking this will be handled by the `x/subspaces` module